quotedprintable.go

```diff
@@ -0,0 +1,67 @@
+package message
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"mime/quotedprintable"
+)
+
+// maxQuotedPrintableLineLength defines the longest line accepted. Longer
+// lines are splitted by inserting soft breakes.
+// By RFC is allowed line with 76 characters long only. Some messages in
+// the wild do not respect this rule. `mime/quotedprintable` can support
+// up to bufio buffer, which is 4096 bytes. Longer lines than what RFC
+// defines but shorter than bufio buffer are kept without modification
+// to not modify message when not neccessary.
+// Note that soft brake has to continue with line ending, i.e., `Read`
+// operation cannot fill the whole quotedprintable buffer with the line
+// without line ending. That is reported as error too. That's why this
+// lenght is two bytes (to include \r\n) shorter then the buffer.
+const maxQuotedPrintableLineLength = 4093
```

Comment:
that's a magic constant not actually guaranteed by the bufio package, you can't rely on it not changing.
That value is neither documented nor considered stable by the docs.

If you do want to have that, you need to force bufio to use a buffer size of that specific length, via bufio.NewReaderSize

Reply:
True, bufio package can change this number between releases. I will use `NewReaderSize` instead to make sure the size of buffer.

Reply:
Actually, this is not that true. The problem is not buffer here but the buffer used by mime/quoteprintable. It creates default bufio reader and that can be changed any time and break this fix. So the magic number has to stay either way. But I will add it for completeness.

Reply:
have a look at Emersion's comment

Reply:
I looked. This should not do unbound allocations and I don't see connection with your comment.

The magic number has to stay but it can still be broken if mime/quoteprintable changes the underlying buffer. There is nothing I can do about it. But I used the sized buffer here and that simplified the implementation a lot.